test/junit/src/test/scala/strawman/collection/SeqTest.scala

```diff
@@ -63,4 +63,20 @@ class SeqTest {
     assertEquals(1, Vector(0, 1, 2, 0, 1, 2).lastIndexOfSlice(Vector(1, 2), end = 3))
     assertEquals(-1, List(0, 1).lastIndexOfSlice(List(1, 2)))
   }
+
+  @Test
+  def hasCorrectDiff(): Unit = {
+    val s1 = Seq(1, 2, 3, 4, 5)
+    val s2 = Seq(1, 3, 5, 7, 9)
+
+    assertEquals(Seq(2, 4), s1.diff(s2))
+  }
+
+  @Test
+  def hasCorrectIntersect(): Unit = {
+    val s1 = Seq(1, 2, 3, 4, 5)
+    val s2 = Seq(1, 3, 5, 7, 9)
+
+    assertEquals(Seq(1, 3, 5), s1.intersect(s2))
+  }
```

Comment:
Are these tests related with the changes of this PR?

Reply:
I added these tests because of this change: https://github.com/scala/collection-strawman/pull/444/files#diff-d2679eddf2a5dee23263f82189f93c81 . It turns out that it was not necessary at all as the anonymous class is only used internally. The behavior is the same though.